src/x86/avx.rs

```diff
@@ -484,6 +484,152 @@ pub unsafe fn _mm256_xor_ps(a: f32x8, b: f32x8) -> f32x8 {
+pub const _CMP_EQ_UQ: u8 = 0x08;
+// Not-greater-than-or-equal (unordered, signaling)
+pub const _CMP_NGE_US: u8 = 0x09;
+// Not-greater-than (unordered, signaling)
+pub const _CMP_NGT_US: u8 = 0x0a;
+// False (ordered, non-signaling)
+pub const _CMP_FALSE_OQ: u8 = 0x0b;
+// Not-equal (ordered, non-signaling)
+pub const _CMP_NEQ_OQ: u8 = 0x0c;
+// Greater-than-or-equal (ordered, signaling)
+pub const _CMP_GE_OS: u8 = 0x0d;
+// Greater-than (ordered, signaling)
+pub const _CMP_GT_OS: u8 = 0x0e;
+// True (unordered, non-signaling)
+pub const _CMP_TRUE_UQ: u8 = 0x0f;
+// Equal (ordered, signaling)
+pub const _CMP_EQ_OS: u8 = 0x10;
+// Less-than (ordered, non-signaling)
+pub const _CMP_LT_OQ: u8 = 0x11;
+// Less-than-or-equal (ordered, non-signaling)
+pub const _CMP_LE_OQ: u8 = 0x12;
+// Unordered (signaling)
+pub const _CMP_UNORD_S: u8 = 0x13;
+// Not-equal (unordered, signaling)
+pub const _CMP_NEQ_US: u8 = 0x14;
+// Not-less-than (unordered, non-signaling)
+pub const _CMP_NLT_UQ: u8 = 0x15;
+// Not-less-than-or-equal (unordered, non-signaling)
+pub const _CMP_NLE_UQ: u8 = 0x16;
+// Ordered (signaling)
+pub const _CMP_ORD_S: u8 = 0x17;
+// Equal (unordered, signaling)
+pub const _CMP_EQ_US: u8 = 0x18;
+// Not-greater-than-or-equal (unordered, non-signaling)
+pub const _CMP_NGE_UQ: u8 = 0x19;
+// Not-greater-than (unordered, non-signaling)
+pub const _CMP_NGT_UQ: u8 = 0x1a;
+// False (ordered, signaling)
+pub const _CMP_FALSE_OS: u8 = 0x1b;
+// Not-equal (ordered, signaling)
+pub const _CMP_NEQ_OS: u8 = 0x1c;
+// Greater-than-or-equal (ordered, non-signaling)
+pub const _CMP_GE_OQ: u8 = 0x1d;
+// Greater-than (ordered, non-signaling)
+pub const _CMP_GT_OQ: u8 = 0x1e;
+// True (unordered, signaling)
+pub const _CMP_TRUE_US: u8 = 0x1f;
+
+/// Compare packed double-precision (64-bit) floating-point
+/// elements in `a` and `b` based on the comparison operand
+/// specified by `imm8`.
+#[inline(always)]
+#[target_feature = "+avx,+sse2"]
+#[cfg_attr(test, assert_instr(vcmpeqpd, imm8 = 0))] // TODO Validate vcmppd
+pub unsafe fn _mm_cmp_pd(a: f64x2, b: f64x2, imm8: u8) -> f64x2 {
+    macro_rules! call {
+        ($imm8:expr) => { vcmppd(a, b, $imm8) }
+    }
+    constify_imm6!(imm8, call)
+}
+
+/// Compare packed double-precision (64-bit) floating-point
+/// elements in `a` and `b` based on the comparison operand
+/// specified by `imm8`.
+#[inline(always)]
+#[target_feature = "+avx"]
+#[cfg_attr(test, assert_instr(vcmpeqpd, imm8 = 0))] // TODO Validate vcmppd
+pub unsafe fn _mm256_cmp_pd(a: f64x4, b: f64x4, imm8: u8) -> f64x4 {
+    macro_rules! call {
+        ($imm8:expr) => { vcmppd256(a, b, $imm8) }
+    }
+    constify_imm6!(imm8, call)
+}
+
+/// Compare packed single-precision (32-bit) floating-point
+/// elements in `a` and `b` based on the comparison operand
+/// specified by `imm8`.
+#[inline(always)]
+#[target_feature = "+avx,+sse"]
+#[cfg_attr(test, assert_instr(vcmpeqps, imm8 = 0))] // TODO Validate vcmpps
```

Comment:
Does this mean that we just need more `assert_instr` attributes? Or should the intrustruction here be different?

Reply:
We need more `assert_instr`: the instruction depends on the comparison operand.